python/paddle/v2/framework/tests/op_test.py

```diff
@@ -413,32 +349,122 @@ def check_grad(self,
                 output_names,
                 in_place=in_place) for input_to_check in inputs_to_check
         ]
-        grad_names = [
-            grad_var_name(input_to_check) for input_to_check in inputs_to_check
-        ]
-
         cpu_place = core.CPUPlace()
-        cpu_analytic_grads = get_gradient(self.scope, self.op, self.inputs,
-                                          self.outputs, grad_names, cpu_place,
-                                          no_grad_set)
+        cpu_analytic_grads = self._get_gradient(inputs_to_check, cpu_place,
+                                                output_names, no_grad_set)
 
-        self.__assert_is_close(numeric_grads, cpu_analytic_grads, grad_names,
-                               max_relative_error,
+        self.__assert_is_close(numeric_grads, cpu_analytic_grads,
+                               inputs_to_check, max_relative_error,
                                "Gradient Check On %s" % str(cpu_place))
 
         if core.is_compile_gpu() and self.op.support_gpu():
             gpu_place = core.GPUPlace(0)
-            gpu_analytic_grads = get_gradient(self.scope, self.op, self.inputs,
-                                              self.outputs, grad_names,
-                                              gpu_place, no_grad_set)
+            gpu_analytic_grads = self._get_gradient(inputs_to_check, gpu_place,
+                                                    output_names, no_grad_set)
 
             self.__assert_is_close(numeric_grads, gpu_analytic_grads,
-                                   grad_names, max_relative_error,
+                                   inputs_to_check, max_relative_error,
                                    "Gradient Check On %s" % str(gpu_place))
 
-            for c_grad, g_grad, name in itertools.izip(
-                    cpu_analytic_grads, gpu_analytic_grads, grad_names):
-                self.assertTrue(
-                    np.allclose(
-                        c_grad, g_grad, atol=1e-4),
-                    "output name: " + name + " has diff")
+    @staticmethod
+    def _create_var_descs_(block, var_dict):
```

Comment:
We now have two interface to create var desc for inputs and outputs, `append_input_output` and `_create_var_descs_`. It's better to unify them.